app/models/chat_channel.rb

```diff
@@ -107,7 +107,11 @@ def chatable_has_custom_fields?
   end
 
   def allowed_user_ids
-    direct_message_channel? ? chatable.user_ids : nil
+    return chatable.user_ids if direct_message_channel?
+
+    staff_memberships =
+      user_chat_channel_memberships.includes(:user).select { |member| member.user.staff? }
+    staff_memberships.empty? ? nil : staff_memberships.map(&:user_id)
```

Comment:
Code golf, I think you can do this for second line:

```
staff_memberships.map(&:user_id).presence
```

Also Im not too concerned about perf here but wouldn't pluck be slightly better?